### PR TITLE
feat: support op-stack based L2 consumer chain

### PR DESCRIPTION
**Pull request by bap2pecs**
## Summary

this is just to consolidate the two work streams. after this, we will all work on top of `base/consumer-chain-support` branch

## Test Plan

```
make test
make test-e2e
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -55,6 +55,7 @@ build-docker:
 .PHONY: build build-docker
 
 test:
+	make mock-gen
```
Comment (gitferry) [inline Makefile:58]: Maybe I was missing something. Why do we want it here?

File: clientcontroller/interface.go [hunk 2]
```diff
@@ -5,15 +5,15 @@ import (
...
 	"github.com/babylonchain/finality-provider/types"
 )
 
 const (
-	babylonConsumerChainName = "babylon"
+	BabylonConsumerChainName = "babylon"
+	EVMConsumerChainName     = "evm"
 )
 
 type ClientController interface {
```
Comment (gitferry) [inline clientcontroller/interface.go:19]: Since this Controller only works for Babylon, why not call it `BabylonController`?

**Review by gitferry — COMMENTED** [review]
One major comment is that maybe we can rename `ClientController` to `BabylonController` as it is only for Babylon. I appologise for not thinking it deep enough in the first look

File: Makefile (hunk 1 above)
Comment (bap2pecs) [reply Makefile:58]: see https://github.com/babylonchain/finality-provider/pull/335 and https://github.com/babylonchain/finality-provider/issues/344

**Comment by bap2pecs** [comment]
thanks for the review @gitferry! the naming of ClientController can be in another PR as it's a major refactor :)

File: Makefile (hunk 1 above)
Comment (gitferry) [reply Makefile:58]: We used to manually run `make mock-gen` and the generated code should be part of PR. If this is automatic in CI, we might miss updating mocks

**Review by gitferry — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (bap2pecs) [reply Makefile:58]: i see. https://github.com/babylonchain/finality-provider/pull/335 actually reflected an issue due to CI doesn't automatically run mock-gen

but agree that putting it in mock-gen seems wrong. i can remove it

**Review by SebastianElvis — APPROVED** [review]
1 less base branch to maintain :100:

File: Makefile (hunk 1 above)
Comment (bap2pecs) [reply Makefile]: updated

**Pull request merged**